### DISCUSSION TITLE
Establish Local Knowledge base for information input

### DISCUSSION THREAD

**Discussion by wangxing7714436** [assafelovic/gpt-researcher#325] [Ideas]
Since most of agents that designed for problem and research tasks use online search engine， I haven't found any
agent add function to find info from pre cleaned local knowledge base. In some industries such as I worked in, the paper or info may be quite difficult to find online, we built our own documents for research purpose. So if you guys can add local document search function that would be a great help. Since local knowledge base Q&A system is quite familiar nowadays, I think it is not very hard to achieve.

My coding skill is poor otherwise I'd like to contribute codes to your project.(learning langchain...)

Many thanks.

**Comment by danieldekay** [comment]
https://docs.gptr.dev/docs/gpt-researcher/tailored-research#research-on-langchain-documents-%EF%B8%8F

Using Langchain you can bring your own documents easily. If you feel adventurous, chatGPT might help you do what you need to do.
